src/dashboard/apigateway/apigateway/biz/gateway/saver.py

```diff
@@ -0,0 +1,122 @@
+#
+# TencentBlueKing is pleased to support the open source community by making
+# 蓝鲸智云 - API 网关(BlueKing - APIGateway) available.
+# Copyright (C) 2017 THL A29 Limited, a Tencent company. All rights reserved.
+# Licensed under the MIT License (the "License"); you may not use this file except
+# in compliance with the License. You may obtain a copy of the License at
+#
+#     http://opensource.org/licenses/MIT
+#
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+# either express or implied. See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# We undertake not to change the open source license (MIT license) applicable
+# to the current version of the project delivered to anyone in the future.
+#
+from typing import Dict, List, Optional
+
+from django.conf import settings
+from pydantic import BaseModel, Field, validator
+
+from apigateway.core.constants import GatewayTypeEnum
+from apigateway.core.models import Gateway
+
+from .gateway import GatewayHandler
+
+
+class GatewayData(BaseModel):
+    name: str = Field(...)
+    description: str = Field(default="")
+    description_en: Optional[str] = Field(default=None)
+    maintainers: List[str] = Field(default_factory=list)
+    status: int = Field(...)
+    is_public: bool = Field(default=False)
+    gateway_type: Optional[GatewayTypeEnum] = Field(default=None)
+    user_config: Optional[Dict] = Field(default=None)
+
+    @validator("gateway_type")
+    def validate_gateway_type(cls, v):  # noqa: N805
+        return GatewayTypeEnum(v) if isinstance(v, int) else v
+
+
+class GatewaySaver:
+    def __init__(
+        self,
+        gateway_id: Optional[int],
+        gateway_data: GatewayData,
```

Comment:
gateway_id to id;
gateway_data to data;

------

Reply:
done